src/main/java/org/ngafid/flights/calculations/HeadingTrackDiff.java

```diff
@@ -0,0 +1,126 @@
+        PreparedStatement query = connection.prepareStatement(sql);
+
+        query.setString(1, HDG_TRK_DIFF);
+        query.setInt(2, this.flight.getId());
+
+        ResultSet resultSet = query.executeQuery();
+        if (resultSet.next()) {
+            return resultSet.getBoolean(1);
+        }
+
+        return false;
+    }
+
+    public static void main(String [] args) {
+        Options options = new Options();
+
+        Option flightIds = new Option("f", "fleet_ids", true, "list of fleet ids to calculate for");
+        flightIds.setRequired(false);
+        flightIds.setArgs(Option.UNLIMITED_VALUES);
+        options.addOption(flightIds);
+
+        CommandLineParser parser = new DefaultParser();
+        HelpFormatter formatter = new HelpFormatter();
+        CommandLine cmd = null;
+        List<Fleet> fleets = null;
+
+        try {
+            cmd = parser.parse(options, args);
+        } catch (ParseException e) {
+            System.out.println(e.getMessage());
+            formatter.printHelp("ExtractFlights", options);
+
+            System.exit(1);
+        }
+
+        Connection connection = Database.getConnection();
+
+        String [] fleetIdStrs = cmd.getOptionValues('f');
+
+        if (fleetIdStrs == null || fleetIdStrs.length <= 0) {
+            fleets = Fleet.getAllFleets(connection);
+        }
+
+        try {
+            for (Fleet fleet : fleets) {
+                LOG.info("Calculating hdg/trk diff for fleet: " + fleet.getName());
+                int fleetId = fleet.getId();
+                // This may be slow for large fleets like UND 
+                List<Flight> flights = Flight.getFlights(connection, fleetId);
+
+                for (Flight flight : flights) {
+                    List<String> missingParams = flight.checkCalculationParameters(HDG_TRK_DEPENDENCIES);
+                    if (missingParams.isEmpty()) {
+                        HeadingTrackDiff calculation = new HeadingTrackDiff(flight, connection);
+                        CalculatedDoubleTimeSeries hdgTrakDiff = new CalculatedDoubleTimeSeries(connection, HDG_TRK_DIFF, "degrees", true, flight);
+                        if (!calculation.existsInDB(connection)) {
+                            hdgTrakDiff.create(calculation);
+                            hdgTrakDiff.updateDatabase(connection, flight.getId());
+                        } else {
+                            LOG.info("Already calculated for flight " + flight.toString());
+                        }
+                    } else {
+                        //Cant be calculated. 
+                        LOG.severe("Skipping flight " + flight.toString());
+                        LOG.severe("Missing columns: " + missingParams.toString());
+                    }
+                }
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+            System.exit(1);
+        } 
+
+        LOG.info("All done!");
+        System.exit(0);
```

Comment:
Is the sys exit necessary here?

Reply:
I find that it can be. Sometimes the maven process doesn't exit immediately on its own, so the script can hang for a bit. This just expedites the termination of the process so the OS can worry about other things (and so that there isn't a locked process in your terminal 🙂).